docs/user/InstructionsForAutoTuner.md

```diff
@@ -83,21 +100,25 @@ For FastRoute you can use:
   - Global route random seed. This will create a copy of `_FR_FILE_PATH` and modify the global route random seed.
 
 
-
-
 ## How to use
 
 ### General Information
 
 `distributed.py` scripts handles sweeping and tuning of ORFS parameters.
 
-For both sweep and tune modes <mode>:
-```shell
-python3 distributed.py -h
-```
+In the sweeping mode, we manually specify the parameters search space
+in the `json`; whereas in the tuning mode - we will use Ray's Tune 
+feature to automatically optimise hyperparameters using one of the 
+algorithms listed above with autotuner `json` file. These two files 
+are of very different format, and you may refer to the examples below
+for more information. 
```

Comment:
```suggestion
The `distributed.py` script supports two modes of operation: `sweep`, where every possible parameter combination in the search space is tested, and `tune`, where we use Ray's Tune feature to intelligently search the space and optimise hyperparameters using one of the algorithms listed above. The `sweep` mode is useful when we want to isolate or test a single or very few parameters. On the other hand, `tune` is more suitable for finding the best combination of a complex and large number of flow parameters. Both modes rely on user-specified search space that is defined by a `.json` file, they use the same syntax and format, though some features may not be available for sweeping.
```

Reply:
I think this is a more accurate description of the difference. @dralabeing please check if the wording is clear.

Reply:
@vvbandeira . I have a few comments:
1. Both sweep and tune use Ray correct? Also are they both automatic in terms of search? Or is sweep manual based on the full search space defined by the user?
2. What is the difference between the search space by sweep and that of tune? Is one faster or slower? Similar to one but what is the speed impact. 
3. From a  use model perspective , it implies that sweep is faster than tune but may not give accurate results. Is this correct?
4. Why use sweep at all if it does not search intelligently across the design space?
I think we need to clarify usage a bit more along these lines. Also use American english, optimize vs optimise etc.

Reply:
1a. Yes, both `sweep` and `tune` use Ray either for job distribution (both) or search (only tune).
1b. `tune` is ML search on user-defined search space. `sweep` is a "brute force" for-loop-style that goes over all the same kind of user-defined search space. In all cases, the user manually defines the search space via a config `.json` file.
2. As mentioned above, you can think of `tune` as a smart search and `sweep` as a "brute force" exploration (no search is involved).
3. `tune` and `sweep` serve different purposes; we cannot compare them apples-to-apples. `tune` is used when the user wants to know the answer to "What are the values for each parameter in my search space to get the best results from ORFS". While the `sweep` would normally be used to answer, "What happens to my design as the parameter X value is set to {1, 2, ..., 100}?"
4. `sweep` is useful, especially for people developing code in OpenROAD, to find good defaults for knobs or in cases where it is very important that you squeeze and get the absolute best of ORFS regardless of the amount of time and CPU needed.

Reply:
@dralabeing I have added some text to explain the role of Ray in both use cases. Should Jack or I draft a complete example about the intended usage of `sweep` and `tune`?

Reply:
Thanks @vvbandeira In 3 above, there is still a qualitative analysis designers make to determine the impact on the design when parameters are swept. How is this analysis done without applying a cost function?  Is this through manual analysis alone i.e viewing each sweep experiment through timing, area, congestion?  Also, what is Ray doing for the sweep - is it simply scheduling and job management without the search? 

I am not entirely clear about the use model and value of sweep vs tune. Let's discuss offline before we finalize doc changes.